php/libraries/UserPermissions.class.inc

```diff
@@ -169,7 +169,9 @@ class UserPermissions
             );
         }
         foreach ($permissions as $p) {
-            if ($this->_permissionTrueAndValid($p)) {
+            if ($this->_permissionTrueAndValid($p)
```

Comment:
Why not just change this to hasPermission where it's already handled? I can't imagine any situation where you'd want hasAnyPermission and hasPermission to have different boundary permissions that would justify bypassing hasPermission.

Reply:
nice